vercel/resource_project_environment_variable.go

```diff
@@ -0,0 +1,239 @@
+				Type:          types.StringType,
+			},
+			"team_id": {
+				Optional:      true,
+				Description:   "The ID of the Vercel team",
+				PlanModifiers: tfsdk.AttributePlanModifiers{resource.RequiresReplace()},
+				Type:          types.StringType,
+			},
+			"id": {
+				Description:   "The ID of the environment variable",
+				Type:          types.StringType,
+				PlanModifiers: tfsdk.AttributePlanModifiers{resource.UseStateForUnknown()},
+				Computed:      true,
+			},
+		},
+	}, nil
+}
+
+// NewResource instantiates a new Resource of this ResourceType.
+func (r resourceProjectEnvironmentVariableType) NewResource(_ context.Context, p provider.Provider) (resource.Resource, diag.Diagnostics) {
+	return resourceProjectEnvironmentVariable{
+		p: *(p.(*vercelProvider)),
+	}, nil
+}
+
+type resourceProjectEnvironmentVariable struct {
+	p vercelProvider
+}
+
+// Create will create a new environment variable for a Vercel project.
+// This is called automatically by the provider when a new resource should be created.
+func (r resourceProjectEnvironmentVariable) Create(ctx context.Context, req resource.CreateRequest, resp *resource.CreateResponse) {
+	if !r.p.configured {
+		resp.Diagnostics.AddError(
+			"Provider not configured",
+			"The provider hasn't been configured before apply. This leads to weird stuff happening, so we'd prefer if you didn't do that. Thanks!",
+		)
+		return
+	}
+
+	var plan ProjectEnvironmentVariable
+	diags := req.Plan.Get(ctx, &plan)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	response, err := r.p.client.UpsertEnvironmentVariable(ctx, plan.ProjectID.Value, plan.TeamID.Value, plan.toUpsertEnvironmentVariableRequest())
+	if err != nil {
+		resp.Diagnostics.AddError(
+			"Error creating project environment variable",
+			"Could not create project environment variable, unexpected error: "+err.Error(),
+		)
+		return
+	}
+
+	result := convertResponseToProjectEnvironmentVariable(response, plan)
+
+	tflog.Trace(ctx, "created project environment variable", map[string]interface{}{
+		"id":         result.ID.Value,
+		"team_id":    result.TeamID.Value,
+		"project_id": result.ProjectID.Value,
+	})
+
+	diags = resp.State.Set(ctx, result)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+}
+
+// Read will read an environment variable of a Vercel project by requesting it from the Vercel API, and will update terraform
+// with this information.
+func (r resourceProjectEnvironmentVariable) Read(ctx context.Context, req resource.ReadRequest, resp *resource.ReadResponse) {
+	var state ProjectEnvironmentVariable
+	diags := req.State.Get(ctx, &state)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	out, err := r.p.client.GetEnvironmentVariable(ctx, state.ProjectID.Value, state.TeamID.Value, state.ID.Value)
+
+	if client.NotFound(err) {
+		resp.State.RemoveResource(ctx)
+		return
+	}
+
+	if err != nil {
```

Comment:
nit: 

I'd probably remove the whitespace between the operation and it's error handling. This just gives a subtle hint that they are related as it keeps them logically grouped.
So: 
```suggestion
	out, err := r.p.client.GetEnvironmentVariable(ctx, state.ProjectID.Value, state.TeamID.Value, state.ID.Value)
	if client.NotFound(err) {
		resp.State.RemoveResource(ctx)
		return
	}
	if err != nil {
```